### ISSUE TITLE
Integration codyze

### ISSUE THREAD

**Issue by MStumpp** [secguro/secguro-cli#106]
- https://github.com/Fraunhofer-AISEC/codyze
- Wie werden Regeln in codyze beschrieben? coko-dsl?

**Comment by m1cm1c** [comment]
sowohl die version des neuesten releases als auch die aktuell version auf master funktionieren bei mir nicht. ich hab jetzt erst mal [ein issue angelegt, das sie leicht lösen können](https://github.com/Fraunhofer-AISEC/codyze/issues/832). ich erhoffe mir, über ihre antwort darauf herauszufinden, ob ich versuchen soll, die version auf master oder die version des neuesten releases zum laufen zu bringe

**Comment by m1cm1c** [comment]
> Wie werden Regeln in codyze beschrieben? coko-dsl?

ja

beispiel: https://github.com/Fraunhofer-AISEC/codyze/blob/c61e61ee0776b326c0b3503667d8b9b3e94fd2ea/codyze-specification-languages/coko/coko-dsl/src/test/resources/javaimpl.codyze.kts

dokumentation: https://www.codyze.io/Coko/rules/

**Comment by m1cm1c** [comment]
hast du das gefühl, dass es möglich ist, hinreichend generische codyze-regeln aufzustellen, so dass sie projektübergreifend bedeutungsvoll sind?

**Comment by MStumpp** [comment]
Mit "projektübergreifend" ist Code-Stack übergreifend gemeint?
In manchen Fällen vermutlich ja. Ich gehe aber davon aus das Probleme existieren, die nur in einem (oder verwandten) Code-Stacks vorkommen.
Eine gute DSL sollte CodeStack spezifische Probleme unterstützen. 

Gibt es in dem Repo eine Sammlung von definierten Regeln?

**Comment by m1cm1c** [comment]
Ich verstehe nicht, was du mit "Code-Stack" meinst.

> Gibt es in dem Repo eine Sammlung von definierten Regeln?

Meines Wissens nach gibt es nur sehr magere Beispielregeln.

**Comment by MStumpp** [comment]
> Ich verstehe nicht, was du mit "Code-Stack" meinst.

Java, Go, Python....

**Comment by m1cm1c** [comment]
> Mit "projektübergreifend" ist Code-Stack übergreifend gemeint?

Nein. Ich bin mir nicht sicher, ob viel Projekt-Übergreifendes damit möglich ist, bis auf "diese Library-Funktion ist deprecated", was auch schon mit Semgrep möglich ist (oder normalem grep, oder auch bereits in viele Language Servers integriert).

**Comment by MStumpp** [comment]
- [ ] Run codyze natively ohne VM (https://kotlinlang.org/docs/native-overview.html)
- 2 beispielhafte Regeln erstellt mit Coko DSL (DSL Anpassungen ggfs. erforderlich)
   - [ ] Regel die man auch mit grep lösen könnte
   - [ ] Regel die von CPGs profitiert
